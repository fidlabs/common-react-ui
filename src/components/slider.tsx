"use client";

import * as SliderPrimitive from "@radix-ui/react-slider";
import * as React from "react";

import { cn } from "@/lib/utils";

const Slider = React.forwardRef<
  React.ElementRef<typeof SliderPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof SliderPrimitive.Root>
>(({ className, ...props }, ref) => (
  <SliderPrimitive.Root
    ref={ref}
    className={cn(
      "group relative flex w-full touch-none select-none items-center aria-disabled:cursor-not-allowed",
      className
    )}
    {...props}
  >
    <SliderPrimitive.Track className="relative h-2 w-full grow overflow-hidden rounded-full bg-gray-500/50">
      <SliderPrimitive.Range className="absolute h-full bg-dodger-blue group-aria-disabled:bg-gray-500" />
    </SliderPrimitive.Track>
    <SliderPrimitive.Thumb className="block h-5 w-5 rounded-full border-2 border-dodger-blue bg-white ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-dodger-blue focus-visible:ring-offset-2 disabled:pointer-events-none group-aria-disabled:border-gray-500" />
  </SliderPrimitive.Root>
));
Slider.displayName = SliderPrimitive.Root.displayName;

export { Slider };
