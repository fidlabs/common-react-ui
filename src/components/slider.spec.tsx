import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Slider } from "./slider";

describe("Slider component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Slider defaultValue={[50]} max={100} step={1} />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders correctly while disabled", () => {
    const { container } = render(
      <Slider defaultValue={[50]} max={100} step={1} />
    );
    expect(container).toMatchSnapshot();
  });
});
