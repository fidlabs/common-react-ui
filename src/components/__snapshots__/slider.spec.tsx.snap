// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Slider component > renders correctly 1`] = `
<div>
  <span
    aria-disabled="false"
    class="group relative flex w-full touch-none select-none items-center aria-disabled:cursor-not-allowed"
    data-orientation="horizontal"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-gray-500/50"
      data-orientation="horizontal"
    >
      <span
        class="absolute h-full bg-dodger-blue group-aria-disabled:bg-gray-500"
        data-orientation="horizontal"
        style="left: 0%; right: 50%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(50% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        class="block h-5 w-5 rounded-full border-2 border-dodger-blue bg-white ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-dodger-blue focus-visible:ring-offset-2 disabled:pointer-events-none group-aria-disabled:border-gray-500"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</div>
`;

exports[`Slider component > renders correctly while disabled 1`] = `
<div>
  <span
    aria-disabled="false"
    class="group relative flex w-full touch-none select-none items-center aria-disabled:cursor-not-allowed"
    data-orientation="horizontal"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-gray-500/50"
      data-orientation="horizontal"
    >
      <span
        class="absolute h-full bg-dodger-blue group-aria-disabled:bg-gray-500"
        data-orientation="horizontal"
        style="left: 0%; right: 50%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(50% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        class="block h-5 w-5 rounded-full border-2 border-dodger-blue bg-white ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-dodger-blue focus-visible:ring-offset-2 disabled:pointer-events-none group-aria-disabled:border-gray-500"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</div>
`;
